src/hcf-release/src/acceptance-tests-brain/test-scripts/008_cf_usb_test.sh

```diff
@@ -0,0 +1,199 @@
+
+MYSQL_USER=root
+MYSQL_PASS=testpass
+
+SERVER_APP=mysql
+
+SIDECAR_API_KEY=secret-key
+SIDECAR_APP=msc
+
+SERVICE_TYPE=my-service
+SERVICE_INSTANCE=my-db
+
+## # # ## ### Test-specific code ### ## # #
+
+function test_cleanup() {
+    trap "" EXIT ERR
+    set +o errexit
+
+    # Reverse order of creation ...
+    # - service instance
+    # - service type = usb endpoint
+    # - msc sidecar app
+    # - mysql server app
+    # - security groups
+    # - tcp routing
+    # - temp directory
+
+    cf delete-service -f ${SERVICE_INSTANCE}
+    yes | cf usb delete-driver-endpoint ${SERVICE_TYPE}
+    cf delete -f ${SIDECAR_APP}
+    cf delete -f ${SERVER_APP}
+    cf unbind-running-security-group internal-services-workaround
+    cf unbind-staging-security-group internal-services-workaround
+    cf delete-shared-domain -f ${CF_TCP_DOMAIN}
+
+    rm -rf "${TMP}"
+
+    set -o errexit
+    login_cleanup
+}
+trap test_cleanup EXIT ERR
+
+# --(0)-- Initialize tcp routing
+
+cf delete-shared-domain -f ${CF_TCP_DOMAIN} || true
+cf create-shared-domain    ${CF_TCP_DOMAIN} --router-group default-tcp
+cf update-quota default --reserved-route-ports -1
+
+# --(0.1) -- Initialize a security group to allow for inter-app comms
+# Attention: This SG opens the entire internal kube service network.
+
+#echo > ${TMP}/internal-services.json '[{ "destination": "192.168.77.77/32", "protocol": "all" }]'
+echo > ${TMP}/internal-services.json '[{ "destination": "0.0.0.0/0", "protocol": "all" }]'
+
+cf create-security-group       internal-services-workaround ${TMP}/internal-services.json
+cf bind-running-security-group internal-services-workaround
+cf bind-staging-security-group internal-services-workaround
+
+## --(1)-- Create and configure the mysql server
+
+cf push --no-start --no-route --health-check-type none ${SERVER_APP} -o mysql/mysql-server
+cf map-route ${SERVER_APP} ${CF_TCP_DOMAIN} --random-port | tee ${TMP}/mysql
+cf set-env   ${SERVER_APP} MYSQL_ROOT_PASSWORD ${MYSQL_PASS}
+cf set-env   ${SERVER_APP} MYSQL_ROOT_HOST '%'
+cf start     ${SERVER_APP}
+
+MYSQL_PORT=$(get_port mysql)
+
+wait_on_port ${MYSQL_PORT}
+
+## --(2)-- Create and configure the mysql client sidecar for usb.
+
+cf push ${SIDECAR_APP} --no-start -o splatform/cf-usb-sidecar-dev-mysql
+
+# Use a secret key that will be used by the USB to talk to your
+# sidecar, and set the connection parameters for the mysql client
+# sidecar so that it can talk to the mysql server from the previous
+# step.
+cf set-env ${SIDECAR_APP} SIDECAR_API_KEY ${SIDECAR_API_KEY}
+cf set-env ${SIDECAR_APP} SERVICE_MYSQL_HOST ${CF_TCP_DOMAIN}
+cf set-env ${SIDECAR_APP} SERVICE_MYSQL_PORT ${MYSQL_PORT}
+cf set-env ${SIDECAR_APP} SERVICE_MYSQL_USER ${MYSQL_USER}
+cf set-env ${SIDECAR_APP} SERVICE_MYSQL_PASS ${MYSQL_PASS}
+cf start   ${SIDECAR_APP}
+
+# --(3)-- Create a driver endpoint to the mysql sidecar (== service type)
+# Note that the -c ":" is required as a workaround to a known issue
```

Comment:
_What_ known issue? (It would be nice to write that down so we know when to remove the workaround)

Reply:
I have no idea. This is code and comment copied from the original test.

Reply:
Do we know who wrote the original test ?